### PR TITLE
Move wiki related funtions from models to services/wiki

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

File: services/wiki/wiki.go [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/sync"
+
+	"github.com/unknwon/com"
+)
+
+var (
+	reservedWikiNames = []string{"_pages", "_new", "_edit", "raw"}
```
Comment (6543) [inline services/wiki/wiki.go:24]: if we use a map["string"]bool lookup could be faster?

like zeripath mentioned in https://github.com/go-gitea/gitea/pull/9220#discussion_r355133946

Comment (6543) [reply services/wiki/wiki.go:24]: or at least use `util.IsStringInSlice` but i find the map version more elegant

Comment (lunny) [reply services/wiki/wiki.go:24]: @6543 It will not because the items is too less.

Comment (6543) [reply services/wiki/wiki.go:24]: ok so then i would sugessd util.IsStringInSlice simply because of code redundancy @lunny

Comment (6543) [reply services/wiki/wiki.go:24]:
```diff
index b7a2b95f4..59c681164 100644
--- a/services/wiki/wiki.go
+++ b/services/wiki/wiki.go
@@ -15,6 +15,7 @@ import (
        "code.gitea.io/gitea/modules/git"
        "code.gitea.io/gitea/modules/log"
        "code.gitea.io/gitea/modules/sync"
+       "code.gitea.io/gitea/modules/util"
 
        "github.com/unknwon/com"
 )
@@ -25,11 +26,9 @@ var (
 )
 
 func nameAllowed(name string) error {
-       for _, reservedName := range reservedWikiNames {
-               if name == reservedName {
-                       return models.ErrWikiReservedName{
-                               Title: name,
-                       }
+       if util.IsStringInSlice(name, reservedWikiNames) {
+               return models.ErrWikiReservedName{
+                       Title: name,
                }
        }
        return nil
```

Comment (6543) [reply services/wiki/wiki.go]: Optimize could be don later ...

**Comment by 6543** [comment]
Wow - thanks for optimisation :D

**Review by 6543 — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

**Comment by 6543** [comment]
Think ths should go into 1.11.0 ... so we dint differ to mouch with codebase from 1.11.0 to 1.12.0

**Review by lafriks — APPROVED** [review]

**Comment by zeripath** [comment]
Make lg-tm work

**Pull request merged**